### PR TITLE
feat: Refactor/update kafka v5

### PR DESCRIPTION
**Pull request by ovaleanu**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Refactor code to EKS blueprints v5

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: streaming/kafka/cleanup.sh [hunk 1]
```diff
@@ -5,24 +5,6 @@ set -o pipefail
 read -p "Enter the region: " region
 export AWS_DEFAULT_REGION=$region
 
-targets=(
```
Comment (vara-bonthu) [inline streaming/kafka/cleanup.sh:8]: I think we need to keep this to ensure Kafka and add-ons are deleted first before deleting the EKS and VPC

targets=(
  "<Any kafka cluster resources>"
  "module.kubernetes_data_addons"
  "module.eks_blueprints_kubernetes_addons"
)

File: streaming/kafka/variables.tf [hunk 2]
```diff
@@ -22,14 +22,9 @@ variable "vpc_cidr" {
...
-  description = "Public Subnets CIDRs. 4094 IPs per Subnet"
-  default     = ["10.1.192.0/20", "10.1.208.0/20", "10.1.224.0/20"]
-  type        = list(string)
-}
 
-variable "private_subnets" {
-  description = "Private Subnets CIDRs. 16382 IPs per Subnet"
-  default     = ["10.1.0.0/18", "10.1.64.0/18", "10.1.128.0/18"]
-  type        = list(string)
+variable "kafka_namespace" {
```
Comment (vara-bonthu) [inline streaming/kafka/variables.tf:26]: see my above comment. We can make this as a local variable

**Review by vara-bonthu — COMMENTED** [review]
Few minor comments otherwise its good 👍🏼

File: streaming/kafka/addons.tf [hunk 3]
```diff
@@ -1,234 +1,207 @@
...
-module "vpc_cni_irsa" {
-  source                = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
-  version               = "~> 5.14"
-  role_name             = format("%s-%s", local.name, "vpc-cni")
+
+module "vpc_cni_ipv4_irsa" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 5.20"
+
+  role_name             = "${module.eks.cluster_name}-vpc-cni-ipv4"
```
[the remark this answers is not in the document]
Comment (ovaleanu) [reply streaming/kafka/addons.tf:31]: Changed

File: streaming/kafka/addons.tf [hunk 4]
```diff
@@ -1,234 +1,207 @@
...
-}
+  #---------------------------------------
+  # Amazon EKS Managed Add-ons
+  #---------------------------------------
+  eks_addons = {
+    aws-ebs-csi-driver = {
+      service_account_role_arn = module.ebs_csi_driver_irsa.iam_role_arn
+    }
+    coredns = {
+      most_recent = true
```
[the remark this answers is not in the document]
Comment (ovaleanu) [reply streaming/kafka/addons.tf:62]: Removed

File: streaming/kafka/addons.tf [hunk 5]
```diff
@@ -1,234 +1,207 @@
...
-  #    wait             = false # This is critical setting. Check this issue -> https://github.com/hashicorp/terraform-provider-helm/issues/683
-  values = [templatefile("${path.module}/helm-values/kafka-values.yaml", {
-    operating_system = "linux"
-    node_group_type  = "core"
-  })]
-  description = "Strimzi - Apache Kafka on Kubernetes"
-  depends_on  = [module.eks.eks_managed_node_groups]
+resource "kubernetes_namespace" "kafka_namespace" {
+  metadata {
+    name = var.kafka_namespace
```
[the remark this answers is not in the document]
Comment (ovaleanu) [reply streaming/kafka/addons.tf:166]: Changed to local

File: streaming/kafka/cleanup.sh (hunk 1 above)
Comment (ovaleanu) [reply streaming/kafka/cleanup.sh:8]: Added addons modules

**Review by vara-bonthu — APPROVED** [review]
LGTM 🚀  Thanks @ovaleanu for your great work 🙏🏼

**Pull request merged**